pkgs/os-specific/darwin/apple-sdk/default.nix

```diff
@@ -178,24 +211,29 @@ let
       "/System/Library/Frameworks/${name}.framework/${name}"
     ];
 
+    passthru.frameworkName = name;
     meta = with lib; {
       description = "Apple SDK framework ${name}";
       maintainers = with maintainers; [ copumpkin ];
       platforms   = platforms.darwin;
     };
   };
 
-  tbdOnlyFramework = name: { private ? true }: stdenv.mkDerivation {
+  tbdOnlyFramework = name: { private ? true, deps ? {} }: stdenv.mkDerivation {
     name = "apple-framework-${name}";
     dontUnpack = true;
+    nativeBuildInputs = [ print-reexports ];
     installPhase = ''
       mkdir -p $out/Library/Frameworks/
       cp -r ${darwin-stubs}/System/Library/${lib.optionalString private "Private"}Frameworks/${name}.framework \
         $out/Library/Frameworks
       # NOTE there's no re-export checking here, this is probably wrong
```

Comment:
This looks like a stray comment you probably want to remove to avoid confusion. Checking re-exports *is* what you're adding, right?

I'm not familiar with the way frameworks are handled yet tbh, so I can't really provide a useful review. Maybe @thefloweringash, because they recently introduced the tbd stubs?

Your rationale does sound solid and none of the changes strike me as peculiar though, fwiw.

Reply:
Ooh, yikes! Good catch.